makefiles/pseudomodules.inc.mk

```diff
@@ -1,3 +1,4 @@
+PSEUDOMODULES += architecture
```

Comment:
actually `sys/include` is always included, so why the need for a module?

Reply:
IMO every piece of code in RIOT should belong to a (pseudo-/sub-)module. For header-only stuff, this will at least help group and organize documentation.

IMO it is an implementation detail whether something is header only, or contains C files. Users should IMO not worry about implementation details. Thus, having it as a pseudo module will allow them to not know or care that this is (currently) a header only module. If an application developer does not use `USEMODULE += architecture` that would also work (as long as this remains a header-only module). But I see not reason why I should confuse users adding it with a linker warning.

Reply:
To me it looks like fundamental information that should always be available (it kind of is)
No-op Pseudomodules feel weird IMHO, I'd just keep it simple.